src/main/java/frc/robot/subsystems/vision/PhotonSubsystem.java

```diff
@@ -126,51 +122,36 @@ private boolean checkPoseValidity(EstimatedRobotPose pose) {
     }
   }
 
-  public double getTurretYaw(int id) {
+  public PhotonTrackedTarget getTurretLastResult(int id) {
     turretResult = turretCamera.getLatestResult();
     if (turretResult.hasTargets()) {
       List<PhotonTrackedTarget> targets = turretResult.getTargets();
       for (PhotonTrackedTarget target : targets) {
         if (target.getFiducialId() == id) {
-          lastRecordedYawTime = turretResult.getTimestampSeconds();
-          lastRecordedYaw = target.getYaw();
-          return lastRecordedYaw;
+          lastRecordedTime = turretResult.getTimestampSeconds();
+          return target;
         }
       }
     }
-    if (lastRecordedYawTime != 0
-        && lastRecordedYawTime + Constants.LimelightConstants.APRILTAG_STALE_TIME
-            < Timer.getFPGATimestamp()) {
-      return lastRecordedYaw;
+    return null;
```

Comment:
Should cache a new target if it has one, then should return a cached PhotonTrackedTarget if it is recent enough.